backpack/extensions/backprop_extension.py

```diff
@@ -74,38 +89,56 @@ def set_module_extension(
             )
         self.__module_extensions[module] = extension
 
-    def __get_module_extension(self, module):
+    def __get_module_extension(
+        self, module: Module
+    ) -> Callable[[BackpropExtension, Module, Tuple[Tensor], Tuple[Tensor]], None]:
         module_extension = self.__module_extensions.get(module.__class__)
 
         if module_extension is None:
-
-            if isinstance(module, (Sequential, ReduceTuple)):
-                return no_op
-
             if self.__fail_mode is FAIL_ERROR:
                 raise NotImplementedError(
-                    "Extension saving to {} ".format(self.savefield)
-                    + "does not have an extension for "
-                    + "Module {}".format(module.__class__)
+                    f"Extension saving to {self.savefield} "
+                    "does not have an extension for "
+                    f"Module {module.__class__}"
                 )
             elif self.__fail_mode == FAIL_WARN:
-                warnings.warn(
-                    "Extension saving to {} ".format(self.savefield)
-                    + "does not have an extension for "
-                    + "Module {}".format(module.__class__)
-                )
+                for _ in module.parameters():
+                    warnings.warn(
+                        f"Extension saving to {self.savefield} does not have an "
+                        f"extension for Module {module.__class__} "
+                        f"although the module has parameters"
```

Comment:
This only warns if a module has parameters, although parameter-free modules that don't have an extension should also trigger such a warning.

Reply:
Try recycling the message string.

Reply:
I would disagree, that parameter-free modules should trigger a warning.
All first order extensions operate in warning mode. It is pretty annoying if they throw a warning for every module that has no extension. 
Note: In the old version, there was never a warning because, the mode was misspelled. That is the reason for the dictionary check of the mode.

If you think a warning is the best approach, I will change it.

Reply:
Your explanation makes sense, I approve your version. Good catch with the misspelled `FAIL_MODE`. Could you replace the strings passed to `fail_mode` by all extensions by variables `FAIL_WARN, FAIL_ERROR`?

Reply:
I placed it into the `FirstOrderBackpropExtension`.

Reply:
I think it's better to keep `fail_mode` as argument in BackPACK's extensions that are exposed in the API. Your current change breaks the API for users that modified `fail_mode` in their code

Reply:
Added `fail_mode=FAIL_WARN` to `FirstOrderBackpropExtension`'s arguments.

Reply:
I meant that it should be part of the constructors of `BatchGrad`, etc. (maybe I also don't see the latest version).

Reply:
This would be a new feature since the [current version on development](https://github.com/f-dangel/backpack/blob/2c632c9fd2e54119595fc32745eaec4d3babdb4b/backpack/extensions/firstorder/batch_grad/__init__.py#L60) does not have a `fail_mode` parameter.